protocol/communities/manager.go

```diff
@@ -167,6 +158,47 @@ type HistoryArchiveDownloadTask struct {
 	Cancelled  bool
 }
 
+type HistoryArchiveDownloadTaskInfo struct {
+	TotalDownloadedArchivesCount int
+	TotalArchivesCount           int
+	Cancelled                    bool
+}
+
+type ArchiveContract interface {
```

Comment:
I explain that here https://github.com/status-im/status-go/pull/5257#issuecomment-2152326864

Basically there is more work to be done, and what is currently `TorrentManager` or `TorrentContract` should be split further into a composition of smaller structs / interfaces. Then we need to attempt to reduce the API as much as possible, because the current interface is gigantic and still too tightly bound to other parts of the application.